### PR TITLE
Validate that batch config max size is greater than send size

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes https://github.com/open-telemetry/opentelemetry-collector/issues/3119

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: processor/batchprocessor/README.md [hunk 1]
```diff
@@ -19,9 +19,10 @@ The following configuration options can be modified:
 batch will be sent.
 - `timeout` (default = 200ms): Time duration after which a batch will be sent
 regardless of size.
-- `send_batch_max_size` (default = 0): The maximum number of items in a batch.
- This property ensures that larger batches are split into smaller units.
- By default (`0`), there is no upper limit of the batch size.
+- `send_batch_max_size` (default = 0): The upper limit of the batch size.
```
Comment (rakyll) [inline processor/batchprocessor/README.md:22]: It's not clear from the docs what's the difference between send_batch_size and send_batch_max_size. Maybe we can clarify them a bit more in detail.

**Review by rakyll — APPROVED** [review]

File: processor/batchprocessor/config.go [hunk 2]
```diff
@@ -39,5 +41,8 @@ var _ config.Processor = (*Config)(nil)
 
 // Validate checks if the processor configuration is valid
 func (cfg *Config) Validate() error {
+	if cfg.SendBatchMaxSize > 0 && cfg.SendBatchMaxSize < cfg.SendBatchSize {
+		return errors.New("send_batch_max_size must be larger than send_batch_size")
```
Comment (tigrannajaryan) [inline processor/batchprocessor/config.go:45]:
```suggestion
		return errors.New("send_batch_max_size must be larger or equal to send_batch_size")
```

**Review by tigrannajaryan — APPROVED** [review]

File: processor/batchprocessor/README.md (hunk 1 above)
Comment (bogdandrutu) [reply processor/batchprocessor/README.md:22]: Not sure if now is more clear, happy to get some help

Comment (rakyll) [reply processor/batchprocessor/README.md:22]: This looks great, thanks!

**Pull request merged**